src/test/java/picard/analysis/CollectRnaSeqMetricsTest.java

```diff
@@ -112,6 +115,59 @@ public void testBasic() throws Exception {
         Assert.assertEquals(metrics.PCT_R2_TRANSCRIPT_STRAND_READS, 0.666667);
     }
 
+    @DataProvider(name = "rRnaIntervalsFiles")
+    public static Object[][] rRnaIntervalsFiles() throws IOException {
+        return new Object[][] {
+                {null},
+                {File.createTempFile("tmp.rRna.", ".interval_list")}
+        };
+    }
+
+    @Test(dataProvider = "rRnaIntervalsFiles", expectedExceptions = PicardException.class)
+    public void testNoIntevalsNoFragPercentage(final File rRnaIntervalsFile) throws Exception {
+        final SAMRecordSetBuilder builder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+
+        // Add a header but no intervals
+        if ( rRnaIntervalsFile != null  ) {
+            final IntervalList rRnaIntervalList = new IntervalList(builder.getHeader());
+            rRnaIntervalList.write(rRnaIntervalsFile);
+            rRnaIntervalsFile.deleteOnExit();
+        }
+
+        // Create some alignments
+        final String sequence = "chr1";
+        final String ignoredSequence = "chrM";
+
+        // Set seed so that strandedness is consistent among runs.
+        builder.setRandomSeed(0);
+        final int sequenceIndex = builder.getHeader().getSequenceIndex(sequence);
+        builder.addPair("pair1", sequenceIndex, 45, 475);
+        builder.addFrag("ignoredFrag", builder.getHeader().getSequenceIndex(ignoredSequence), 1, false);
+
+        final File samFile = File.createTempFile("tmp.collectRnaSeqMetrics.", ".sam");
+        samFile.deleteOnExit();
+
+        final SAMFileWriter samWriter = new SAMFileWriterFactory().makeSAMWriter(builder.getHeader(), false, samFile);
+        for (final SAMRecord rec : builder.getRecords()) samWriter.addAlignment(rec);
+        samWriter.close();
+
+        // Generate the metrics.
+        final File metricsFile = File.createTempFile("tmp.", ".rna_metrics");
+        metricsFile.deleteOnExit();
+
+        final String rRnaIntervalsPath = rRnaIntervalsFile != null ? rRnaIntervalsFile.getAbsolutePath() : null;
+        final String[] args = new String[]{
+                "INPUT=" + samFile.getAbsolutePath(),
+                "OUTPUT=" + metricsFile.getAbsolutePath(),
+                "REF_FLAT=" + getRefFlatFile(sequence).getAbsolutePath(),
+                "RIBOSOMAL_INTERVALS=" + rRnaIntervalsPath,
+                "RRNA_FRAGMENT_PERCENTAGE=" + 0.0,
+                "STRAND_SPECIFICITY=SECOND_READ_TRANSCRIPTION_STRAND",
+                "IGNORE_SEQUENCE=" + ignoredSequence
+        };
+        Assert.assertEquals(runPicardCommandLine(args), 0);
+    }
```

Comment:
I don't see a test where it fails if the intervals file is empty and `RRNA_FRAGMENT_PERCENTAGE=0`